Mathlib/CategoryTheory/Limits/Shapes/ZeroMorphisms.lean

```diff
@@ -220,7 +220,7 @@ theorem iff_isSplitMono_eq_zero {X Y : C} (f : X ⟶ Y) [IsSplitMono f] : IsZero
     rw [← Category.id_comp f, h, zero_comp]
   · intro h
     rw [← IsSplitMono.id f]
-    simp [h]
+    simp only [h, zero_comp]
```

Comment:
I think this is a good change, just asking for curiosity: (why) is this needed? Again, feel free to discard.

Reply:
I'm backporting this in #11079.